grails-app/controllers/org/pih/warehouse/core/ErrorsController.groovy

```diff
@@ -113,6 +114,24 @@ class ErrorsController {
         render(view: "/error")
     }
 
+    def handleSqlIntegrityConstraintViolationException() {
+        if (RequestUtil.isAjax(request)) {
+            if (request?.method == HttpMethod.DELETE.name()) {
+                String message = g.message(
+                        code: "errors.existingAssociation.message",
+                        default: "Resource could not be deleted because of an existing association"
+                )
+
+                render([ errorCode: 500, errorMessages: [message]] as JSON)
+                return
+            }
+
+            Throwable root = ExceptionUtils.getRootCause(request.getAttribute('exception'))
+            render([errorCode: 500, errorsMessages: [root.getMessage()]])
```

Comment:
you have a typo here - `errorsMessages` instead of `errorMessages`.
By the way if we have a single message, I'd consider using `errorMessage`.

Reply:
> you have a typo here - errorsMessages instead of errorMessages.

Good catch.

> By the way if we have a single message, I'd consider using errorMessage.

Can you elaborate on that? I assume our convention is to use the "errorMessages" key, so why would it be better to have a duplicate key "errorMessage" that requires the API client to check for values under both keys?

Reply:
@jmiranda we already have `errorMessage` - e.g. look at `handleValidationErrors` - we pass both `errorMessage` and `errorMessages`. Frontend side (error handler) is already prepared for both options, so we wouldn't have to introduce anything new. It's just that if we have a single message (like in this), not an array of messages (like for validation errors), we should pass `errorMessage` in my opinion, because in the end frontend would still destruct this array and make it a single message.